### PR TITLE
1.19 Partial reversion of boats, reorganization of new generation

### PR DESCRIPTION
**Pull request by doofus-01**
This is to revert (partially or fully) recent changes to the old boats, leaving them as fake units, and allow new ships to be free of their influence.

This is the current state of affairs:
Screenshot_20241229_ships

Known To Do:

 - [ ]  ~Coherent plan for unit balance - the precise details don't necessarily need to be finalized here, but a general plan that could affect sprites and animations should be completed.~
 - [ ] ~Transport Ship vs Pirate ship differentiation~
 - [x] Orc ship
 - [ ] ~Elf Ship~
 - [x] Undead Ship
 - [ ] ~Dwarf Ship~
 - [x]  Primitive ship/barge (for saurians, ogres, etc.)

More variety is better (e.g. orcish skiff) but one of each main race makes sense a minimum.

EDIT:  This PR scope was too much, I need to reduce it.  The dropped components will still be addressed, just not here and now.
Screenshot_20250413_
EDIT2: Orcish ships now have the basic requirements.  Descriptions and unit balance are almost nonexistent, but that can be done in a subsequent PR.

### CODE REVIEW

**Comment by Dalas121** [comment]
Thanks for splitting out the new, pretty sprites from the old, fake unit stats. The skiff is also a valuable addition, though to my eye something looks off about the sail.

Seeing the beautiful new boats you added, I'd been finishing up a drastic rebalance for them: [https://github.com/Dalas121/wesnothDalas/tree/ships-carracks-and-raiders](https://github.com/Dalas121/wesnothDalas/tree/ships-carracks-and-raiders). I'll wait until after this PR is merged to continue work on mine.

**Comment by doofus-01** [comment]
> Thanks for splitting out the new, pretty sprites from the old, fake unit stats. The skiff is also a valuable addition, though to my eye something looks off about the sail.

The skiff got screwed up by the pallet cleanup script; something is wrong with the colors in the originals, it will look better before merging.

> Right now the Pirate and Transport ship are very similar. Is there any chance of getting a stronger visual distinction between the two; maybe a skull or stripe on the Pirate's sail?

I agree, they are too similar, it is a legacy of the original fake units, it will be fixed.  The visual differences should reflect the different abilities, so maybe we can determine those first.  I guess Pirate is faster and higher damage, at the expense of HP? 

> Seeing the beautiful new boats you added, I'd been finishing up a drastic rebalance for them: https://github.com/Dalas121/wesnothDalas/tree/ships-carracks-and-raiders. I'll wait until after this PR is merged to continue work on my rebalance to avoid any conflicts, but I wanted to let you know.

Thanks for the heads-up.  It's hard to follow via github commits, we should have some sort of working document to avoid conflicts.  I'll update the OP for now, but maybe we should take the discussion to the forums, so it is visible and under the control of Wesnoth (in case it matters years later).

**Comment by Dalas121** [comment]
>  The visual differences should reflect the different abilities, so maybe we can determine those first. I guess Pirate is faster and higher damage, at the expense of HP?

> It's hard to follow via github commits, we should have some sort of working document to avoid conflicts. I'll update the OP for now, but maybe we should take the discussion to the forums

Having a public conversation on the forums is a good idea. I've created an initial post with some of my ideas: https://forums.wesnoth.org/viewtopic.php?p=695664


> The skiff got screwed up by the pallet cleanup script; something is wrong with the colors in the originals, it will look better before merging.

I wonder if the Skiff would look better with no sail at all? I see it's intended to be a lower-level boat, but the huge sail makes it take up nearly as many pixels as the pirate/transport, and also makes obvious the skiff's different scale.


> Dwarf Ship

Personally, I'm not sure if there's a need for a dwarven ship - not only do they prefer living in caves and mountains, but they also have the Gryphon for small-scale water transport. Leaving them ship-less could be a way to emphasize their differences with the other human-ish races (and would save dev time).


> [frame]
>     image="misc/blank-hex.png:1600"
> [/frame]
One downside of this approach is that slow/stun don't change the unit's visual, nor does the "glow" from preadvance/postadvance. You're probably already aware, but bringing it up just in case it's something you hadn't considered.

**Comment by doofus-01** [comment]
> Having a public conversation on the forums is a good idea. I've created an initial post with some of my ideas: https://forums.wesnoth.org/viewtopic.php?p=695664

Cool, thanks!  I'll wade in later.

> I wonder if the Skiff would look better with no sail or mast at all? I see it's intended to be a lower-level boat, but with its huge sail it takes up nearly as many pixels as the pirate/transport - and this also makes obvious the skiff's different scale.

That is a good point.  I will try to shrink it down, and make it more lateen or crab-claw, so the single sail can span more of the sprite without looking weird.

> Personally, I'm not sure if there's a need for a Dwarvish ship - not only do Dwarves prefer living in caves and mountains, but they also have the Gryphon for small-scale water transport. Leaving them ship-less could be a way to emphasize their differences with the other human-ish races (and would save dev time).

That could be, or there could be some very specialized, very slow ship.  But it makes sense to not make it a high priority.

> One downside of this approach is that slow/stun don't change the unit's appearance, nor does the "glow" from preadvance/postadvance. You're probably already aware, but bringing it up just in case it's something you hadn't considered.

To be honest, I'd forgotten but I think there is a way around that.  Thanks for the reminder.

**Comment by ForestDragon-wesnoth** [comment]
@doofus-01 What do you think about this idea for an elf ship design - instead of being made from cut planks like normal ships, it could be some sort of magical living tree with branches/leaves/etc., carefully molded by magic to be able to be used as a ship, and being able to feed on seawater with its roots to sustain itself (unlike normal trees, but again, magic). The sails could even be giant leaves too.

**Comment by doofus-01** [comment]
Giant leaves as sails might or might not be too far out, but a living ship sounds interesting and worth trying!

**Comment by doofus-01** [comment]
I needed a small project to get back into this, these little non-combat boats seemed appropriate:
Screenshot_20250323_

The screenshot shows some shore-water transitions issues, but those aren't anything new and they are outside this PR.

**Comment by doofus-01** [comment]
This will take forever to complete, so I will scale back the scope of the PR.  Elf ships will be in a different PR, dwarf ships may or may not be a thing, and refinement of all ships will happen after there are concrete assets available to work with.  This PR will just be the undead, orc, and primitive boats.  I will update the OP.

**Comment by doofus-01** [comment]
I will commit this once CI is finished.  The old ships are kept as "obsolete, backwards-compatibility" units rather than fully restored.  Sort of like an old terrain with a "D" on it, they get a sepia IPF.  There are other ways of dealing with that, it can be revised if needed, but I don't think it breaks anything.

**Comment by Wedge009** [comment]
(I haven't read this conversation.)

The Pirate Ship and Transport Ship both have practically the same description as their Galleon counterparts. Is this intentional?

**Comment by doofus-01** [comment]
Sort of.  I didn't have much to say about them, most players won't see the descriptions side-by-side, and it's probably going to get changed anyway.

**Comment by Wedge009** [comment]
Okay. Well, it's going to be visible for 1.19.12 at least (since I noticed these in the translation files), although I see you've just moved these units to 'legacy'.
